File: problema 9/fortran/problema_9.f90
program transposeMatrix
    implicit none
    integer :: matriz(5,5)
    integer :: transpuesta(5,5)
    integer :: i, j

    ! Input elements of the matrix
    print *, "Ingrese los elementos de la matriz (5x5):"
    do i = 1, 5
        do j = 1, 5
            read *, matriz(i, j)
        end do
    end do

    ! Compute the transpose of the matrix
    do i = 1, 5
        do j = 1, 5
            transpuesta(j, i) = matriz(i, j)
        end do
    end do

    ! Display the original matrix
    print *, "Matriz original:"
    do i = 1, 5
        do j = 1, 5
            write(*, '(I3)', advance='no') matriz(i, j)
        end do
        print *, ""
    end do

    ! Display the transposed matrix
    print *, "Matriz transpuesta:"
    do i = 1, 5
        do j = 1, 5
            write(*, '(I3)', advance='no') transpuesta(i, j)
        end do
        print *, ""
    end do

end program transposeMatrix

File: problema 9/c-languaje/problema_9.c
#include <stdio.h>

int main() {
    int matriz[5][5];
    int transpuesta[5][5];
    int i, j;

    // Input elements of the matrix
    printf("Ingrese los elementos de la matriz (5x5):\n");
    for (i = 0; i < 5; i++) {
        for (j = 0; j < 5; j++) {
            scanf("%d", &matriz[i][j]);
        }
    }

    // Compute the transpose of the matrix
    for (i = 0; i < 5; i++) {
        for (j = 0; j < 5; j++) {
            transpuesta[j][i] = matriz[i][j];
        }
    }

    // Display the original matrix
    printf("Matriz original:\n");
    for (i = 0; i < 5; i++) {
        for (j = 0; j < 5; j++) {
            printf("%d\t", matriz[i][j]);
        }
        printf("\n");
    }

    // Display the transposed matrix
    printf("Matriz transpuesta:\n");
    for (i = 0; i < 5; i++) {
        for (j = 0; j < 5; j++) {
            printf("%d\t", transpuesta[i][j]);
        }
        printf("\n");
    }

    return 0;
}